### PR TITLE
Fix #1080

### LINKED ISSUE #1080: metrics_api_SUITE:end_per_group failed for Reason: badarg
MongooseIM version: master
Installed from: source
Erlang/OTP version: 18.3

The test cleanup code metrics_api_SUITE:end_per_group fails for Reason: badarg.  This is not noticed by the travis-test.sh tests, because it does not exit non-zero.  But in my environment (jenkins + cth_surefire) it fails.

You can see the failure in your S3 logs here: http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1033/2182/internal_mnesia.18.3/big/ct_run.test@testing-gce-2e867be6-913c-4bbb-b1c7-5c1a83d80a96.2016-11-03_13.43.46/ejabberd_tests.tests.metrics_api_SUITE.logs/run.2016-11-03_13.54.21/metrics_api_suite.end_per_group.786.html

PR to follow.

### PR DESCRIPTION
**Pull request by kenstir**
This PR addresses #1080 

Proposed changes include:
* change test group finalise method to delete users given the test config, not the atom 'ok'

### CODE REVIEW

File: test/ejabberd_tests/tests/metrics_helper.erl [hunk 1]
```diff
@@ -47,8 +47,8 @@ finalise_by_all_metrics_are_global(Config, false) ->
 finalise_by_all_metrics_are_global(Config, true) ->
     Config1 = lists:keydelete(all_metrics_are_global, 1, Config),
     %% TODO: Refactor once escalus becomes compatible with multiple nodes RPC
-    Config2 = distributed_helper:remove_node_from_cluster(ejabberd_node_utils:mim2(), Config1),
-    escalus:delete_users(Config2, escalus:get_users(?ONLY_GLOBAL_METRICS_GROUP_USERS)).
+    distributed_helper:remove_node_from_cluster(ejabberd_node_utils:mim2(), Config1),
```
Comment (gadgetci) [inline test/ejabberd_tests/tests/metrics_helper.erl]: According to **Elvis**:
> Line 50 is too long:     distributed_helper:remove_node_from_cluster(ejabberd_node_utils:mim2(), Config1),.

File: test/ejabberd_tests/tests/metrics_helper.erl [hunk 2]
```diff
@@ -47,8 +47,8 @@ finalise_by_all_metrics_are_global(Config, false) ->
...
+    escalus:delete_users(Config1, escalus:get_users(?ONLY_GLOBAL_METRICS_GROUP_USERS)).
```
Comment (gadgetci) [inline test/ejabberd_tests/tests/metrics_helper.erl]: According to **Elvis**:
> Line 51 is too long:     escalus:delete_users(Config1, escalus:get_users(?ONLY_GLOBAL_METRICS_GROUP_USERS))..

File: .travis.yml [hunk 3]
```diff
@@ -49,13 +49,14 @@ otp_release:
         - 18.3
 env:
         - PRESET=internal_mnesia DB=mnesia REL_CONFIG=with-all
+ken_disabled_upstream_env:
```
Comment (michalwski) [inline .travis.yml]: Re-enable all the disabled presets for mainstream repo. I understand in your setup you need only one preset. Here in the main repo we need to test all the configurations.

File: .travis.yml [hunk 4]
```diff
@@ -49,13 +49,14 @@ otp_release:
...
         - PRESET=mysql_redis DB=mysql REL_CONFIG="with-mysql with-redis"
         - PRESET=odbc_pgsql_mnesia DB=pgsql REL_CONFIG=with-odbc
         - PRESET=ldap_mnesia DB=mnesia REL_CONFIG=with-none
         - PRESET=riak_mnesia DB=riak REL_CONFIG=with-riak
         - PRESET=dialyzer_only
 
-matrix:
+ken_disabled_upstream_matrix:
```
Comment (michalwski) [inline .travis.yml]: The same thing as re `ken_disabled_upstream_env`

File: test/ejabberd_tests/jenkins.spec [hunk 5]
```diff
@@ -0,0 +1,74 @@
+%% Spec examples:
```
Comment (michalwski) [inline test/ejabberd_tests/jenkins.spec]: We probably don't need `jenkins.spec` file in main repo.

File: tools/jenkins-test.sh [hunk 6]
```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
```
Comment (michalwski) [inline tools/jenkins-test.sh]: Is there any way to merge this file with `tools/travis-test.sh`?

**Comment by kenstir** [comment]
Hi Michal,

Thanks for your helpful comments.  I accidentally submitted this PR with all the changes in my repo, when what I really wanted was just the single commit fe4a97c fixing the issue.  I am sorry for the trouble.  What is better for you, a new PR or an update to this one?

**Comment by michalwski** [comment]
You can rebase and force push this one.

File: test/ejabberd_tests/tests/metrics_helper.erl (hunk 2 above)
Comment (gadgetci) [inline test/ejabberd_tests/tests/metrics_helper.erl]: According to **Elvis**:
> Line 51 is too long:     escalus:delete_users(Config1, escalus:get_users(?ONLY_GLOBAL_METRICS_GROUP_USERS))..

File: test/ejabberd_tests/tests/metrics_helper.erl (hunk 1 above)
Comment (gadgetci) [inline test/ejabberd_tests/tests/metrics_helper.erl]: According to **Elvis**:
> Line 50 is too long:     distributed_helper:remove_node_from_cluster(ejabberd_node_utils:mim2(), Config1),.

**Comment by michalwski** [comment]
Thanks a lot @kenstir!

**Pull request merged**